src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -229,87 +225,15 @@ def compute_num_masked_span(input_length):
     return spec_aug_mask
 
 
-def _median_filter(inputs: torch.Tensor, filter_width: int) -> torch.Tensor:
-    """
-    Applies a median filter of width `filter_width` along the last dimension of the input.
-
-    The `inputs` tensor is assumed to be 3- or 4-dimensional.
-    """
-    if filter_width <= 0 or filter_width % 2 != 1:
-        raise ValueError("`filter_width` should be an odd number")
-
-    pad_width = filter_width // 2
-    if inputs.shape[-1] <= pad_width:
-        return inputs
-
-    # Pad the left and right edges.
-    inputs = nn.functional.pad(inputs, (pad_width, pad_width, 0, 0), mode="reflect")
-
-    # sort() is faster than torch.median (https://github.com/pytorch/pytorch/issues/51450)
-    result = inputs.unfold(-1, filter_width, 1).sort()[0][..., pad_width]
-    return result
-
-
-def _dynamic_time_warping(matrix: np.ndarray):
-    """
-    Measures similarity between two temporal sequences: the input audio and the output tokens. Used to generate
-    token-level timestamps.
-    """
-    output_length, input_length = matrix.shape
-    cost = np.ones((output_length + 1, input_length + 1), dtype=np.float32) * np.inf
-    trace = -np.ones((output_length + 1, input_length + 1), dtype=np.float32)
-
-    cost[0, 0] = 0
-    for j in range(1, input_length + 1):
-        for i in range(1, output_length + 1):
-            c0 = cost[i - 1, j - 1]
-            c1 = cost[i - 1, j]
-            c2 = cost[i, j - 1]
-
-            if c0 < c1 and c0 < c2:
-                c, t = c0, 0
-            elif c1 < c0 and c1 < c2:
-                c, t = c1, 1
-            else:
-                c, t = c2, 2
-
-            cost[i, j] = matrix[i - 1, j - 1] + c
-            trace[i, j] = t
-
-    # backtrace
-    i = trace.shape[0] - 1
-    j = trace.shape[1] - 1
-    trace[0, :] = 2
-    trace[:, 0] = 1
-
-    text_indices = []
-    time_indices = []
-    while i > 0 or j > 0:
-        text_indices.append(i - 1)
-        time_indices.append(j - 1)
-        if trace[i, j] == 0:
-            i -= 1
-            j -= 1
-        elif trace[i, j] == 1:
-            i -= 1
-        elif trace[i, j] == 2:
-            j -= 1
-        else:
-            raise RuntimeError(
-                f"Internal error in dynamic time warping. Unexpected trace[{i}, {j}]. Please file a bug report."
-            )
-
-    text_indices = np.array(text_indices)[::-1]
-    time_indices = np.array(time_indices)[::-1]
-    return text_indices, time_indices
-
-
 class WhisperPositionalEmbedding(nn.Embedding):
     def __init__(self, num_positions: int, embedding_dim: int, padding_idx: Optional[int] = None):
         super().__init__(num_positions, embedding_dim)
 
-    def forward(self, input_ids, past_key_values_length=0):
-        return self.weight[past_key_values_length : past_key_values_length + input_ids.shape[1]]
+    def forward(self, input_ids, past_key_values_length=0, position_ids=None):
```

Comment:
In order to allow batched long-form generation, we need to allow passing the `decoder_position_ids`. This is also needed for speculative decoding